pkg/config/config.go

```diff
@@ -981,6 +981,21 @@ func getMultipleEndpointsWithConfig(config Config) (map[string][]string, error)
 	return keysPerDomain, nil
 }
 
+// ComputeAPIDomain returns a domain to be used for API validation
+func ComputeAPIDomain(domain string) string {
+	apiDomain := ""
+	if parsedDomain, err := url.Parse(domain); err == nil {
+		if _, found := ddURLs[parsedDomain.Host]; found {
+			apiDomain = strings.Replace(domain, "app.", "api.", 1)
+		} else {
+			apiDomain = domain
+		}
+		return apiDomain
+	}
+	log.Debugf("Parsing Warning: The domain %s was unable to be parsed for API Validation, the agent will continue to use this domain for API validation instead.", domain)
```

Comment:
instead of logging something here, let's return make this function return an error (so `(string, error)` instead of `string`), and handle the error in the caller (and the caller can log such a message if this function returns an error).

The benefit of doing this is that it makes this function's behavior better defined and more testable; conversely you should also add a test that checks what happens when the function is called with a non-parseable URL.